splinepy/spline.py

```diff
@@ -1509,32 +1510,52 @@ def proximities(
           Default is False.
           Set search bound to aabb of direct neighbors from the sample grid.
         nthreads: int
+        return_verbose : bool
+          If False, returns only parametric coords
 
         Returns
         --------
         para_coord: (n, para_dim) np.ndarray
+          Parametric coordinates
         phys_coord: (n, dim) np.ndarray
+          (only if return_verbose) respective physical coordinates
         phys_diff: (n, dim) np.ndarray
+          (only if return_verbose) respective cartesian difference to query
         distance: (n, 1) np.ndarray
+          (only if return_verbose) respective 2-norm of difference
         convergence_norm: (n, 1) np.ndarrray
+          (only if return_verbose) Newton residual
         first_derivatives: (n, para_dim, dim) np.ndarray
+          (only if return_verbose) Convergence information
         second_derivatives: (n, para_dim, para_dim, dim) np.ndarray
+          (only if return_verbose) Convergence information
         """
         self._logd("Searching for nearest parametric coord")
 
         queries = utils.data.enforce_contiguous(queries, dtype="float64")
 
-        # so long, so-long-varname
-        igsr = initial_guess_sample_resolutions
-
-        return super().proximities(
-            queries=queries,
-            initial_guess_sample_resolutions=igsr,
-            tolerance=_default_if_none(tolerance, settings.TOLERANCE),
-            max_iterations=max_iterations,
-            aggressive_search_bounds=aggressive_search_bounds,
-            nthreads=_default_if_none(nthreads, settings.NTHREADS),
-        )
+        if return_verbose:
+            return super().proximities(
+                queries=queries,
+                initial_guess_sample_resolutions=_default_if_none(
+                    tolerance, self.control_mesh_resolutions * 2
+                ),
+                tolerance=_default_if_none(tolerance, settings.TOLERANCE),
+                max_iterations=max_iterations,
+                aggressive_search_bounds=aggressive_search_bounds,
+                nthreads=_default_if_none(nthreads, settings.NTHREADS),
+            )
+        else:
+            return super().proximities(
+                queries=queries,
+                initial_guess_sample_resolutions=_default_if_none(
+                    tolerance, self.control_mesh_resolutions * 2
+                ),
+                tolerance=_default_if_none(tolerance, settings.TOLERANCE),
+                max_iterations=max_iterations,
+                aggressive_search_bounds=aggressive_search_bounds,
+                nthreads=_default_if_none(nthreads, settings.NTHREADS),
+            )[0]
```

Comment:
```suggestion
        verbose_info = super().proximities(
                queries=queries,
                initial_guess_sample_resolutions=_default_if_none(
                    initial_guess_sample_resolutions, self.control_mesh_resolutions * 2
                ),
                tolerance=_default_if_none(tolerance, settings.TOLERANCE),
                max_iterations=max_iterations,
                aggressive_search_bounds=aggressive_search_bounds,
                nthreads=_default_if_none(nthreads, settings.NTHREADS),
        )

        if return_verbose:
            return verbose_info
        else:
            return verbose_info[0]
```
maybe non-verbose output can check the distance or convergence_norm against given tolerance and add warning log

Reply:
Good idea. I'll adjust it

Reply:
I updated the code according to your suggestions